File: Algorithms/searching/linear_search.c
#include <stdio.h>

int linearSearch(int arr[], int key, int n)
{
    for (int i = 0; i < n; i++)
    {
        if (arr[i] == key)
            return i;
    }
    return -1;
}

int main()
{
    int arr[] = {1, 3, 2, 0, 5, 4, -1};
    int key = 3, n;
    if (linearSearch(arr, key, n))
        printf("%d found at index %d\n", key, linearSearch(arr, key, n));
    else
        printf("%d not found", key);
    return 0;
}

File: Algorithms/searching/linear_search.py
def search(a, key):
    for i in range(len(a)):
        if a[i] == key:
            return i

    return -1


if __name__ == "__main__":

    a = [1, 3, 2, 0, 5, 4, -1]
    key = 9
    print(f"{key} found at index {search(a, key)}") if search(a, key) != -1 else print(
        f"{key} not found"
    )
